File: algorithm/to-lower-case.py
class Solution:
    def toLowerCase(self, str):
        """
        :type str: str
        :rtype: str
        """
        res = []
        for s in str:
            ord_s = ord(s)
            if 65 <= ord_s <= 90:
                res.append(chr(ord_s + 32))
            else:
                res.append(s)
        return ''.join(res)


if __name__ == '__main__':
    string = 'Lovely'
    res = Solution().toLowerCase(string)
    print(res)

File: Ruby/to-lower-case.rb
# @param {String} str
# @return {String}
def to_lower_case(str)
  res = ""
  str.each_char do |item|
    item_ord = item.ord
    if 65 <= item_ord and item_ord <= 90
      res.concat((item_ord + 32).chr)
    else
      res.concat(item)
    end
  end
  return res
end

def to_lower_case1(str)
  return str.downcase
end

res = to_lower_case1("ABc")
p res